### PR TITLE
[Search] Add opensearch provider 14987

### PR DESCRIPTION
**Pull request by aitherios**
## Why / What

This patch continues the work started by jinnjwu on https://github.com/backstage/backstage/pull/14988 .

Backstage search currently supports the opensearch provider but only in one form. When supplying `aws` to your search provider config it assumes aws managed opensearch. This patch aims to support a hosted solution, for example [docker](https://hub.docker.com/r/opensearchproject/opensearch).

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by camilaibs — COMMENTED** [review]
Hello @aitherios, thanks for contributing to this with @jinnjwu, at first glance you are missing only the changeset file, could you generate it (the instructions can be found in this comment: https://github.com/backstage/backstage/pull/15174#Issuecomment-1346657853)!

Also, @jinnjwu would you like to test it locally? We would appreciate your test, as this is a real case for you! If not possible, I will come back later and try to execute it locally.

File: plugins/search-backend-module-elasticsearch/api-report.md [hunk 1]
```diff
@@ -421,7 +421,7 @@ export interface OpenSearchElasticSearchClientOptions
   // (undocumented)
   nodes?: string | string[] | OpenSearchNodeOptions | OpenSearchNodeOptions[];
   // (undocumented)
-  provider?: 'aws';
+  provider?: 'aws' | 'opensearch';
```
Comment (camilaibs) [inline plugins/search-backend-module-elasticsearch/api-report.md:424]: Could you please update the documentation here?
https://backstage.io/docs/features/search/search-engines#example-configurations

File: plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts [hunk 2]
```diff
@@ -452,6 +454,24 @@ export async function createElasticSearchClientOptions(
...
   }
+  if (config.getOptionalString('provider') === 'opensearch') {
+    const authConfig = config.getConfig('auth');
+    return {
+      provider: 'opensearch',
+      auth: {
+        username: authConfig.getString('username'),
+        password: authConfig.getString('password'),
+      },
+      ...(sslConfig
```
Comment (camilaibs) [inline plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts:465]: It is missing the node config...

Comment (aitherios) [reply plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts:466]: Solved on commit 9de6464aee55e0b9bc978292941d9d87f13baead

**Review by camilaibs — APPROVED** [review]
LGTM 💯

**Pull request merged**